### PR TITLE
chore: component api docs r-s

### PR DESCRIPTION
**Pull request by SiTaggart**
- chore: checkin
- chore: api docs

Description and useful links to discussions / issues / tickets

### Change One

Details of change...

### Change Two

Details of change...

## Checklist

- [ ] Work in a "Draft" branch whilst you are getting feedback to reduce Chromatic costs. Once you are ready for approval, convert the PR to "Ready to review"
- [ ] Ensure all `required` checks have passed
- [ ] Icon changes must have a product design review
- [ ] Website content changes must have a product design review
- [ ] At least two engineers have reviewed any code changes
- [ ] At least one engineering lead has reviewed any core package changes or repository infrastructure
- [ ] Website visual regression testing is run by adding `🕵🏻‍♀️ Run website visual regression` label

### CODE REVIEW

**Review by nkrantz — COMMENTED** [review]
monster

File: .changeset/thick-lamps-knock.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+"@twilio-paste/media-object": patch
+"@twilio-paste/box": patch
+"@twilio-paste/sibling-box": patch
+"@twilio-paste/text": patch
+"@twilio-paste/icons": patch
+"@twilio-paste/style-props": patch
+"@twilio-paste/core": patch
+---
+
+Improved TsDoc type annotations
```
[the remark this answers is not in the document]
Comment (SiTaggart) [reply .changeset/thick-lamps-knock.md:36]: I _knew you would say that_

I'm so lazy!

File: packages/paste-core/components/anchor/type-docs.json [hunk 2]
```diff
@@ -377,7 +377,8 @@
       "type": "\"space0\" | \"space10\" | \"space20\" | \"space30\" | \"space40\" | \"space50\" | \"space60\" | \"space70\" | \"space80\" | \"space90\" | \"space100\" | \"space110\" | \"space120\" | \"space130\" | \"space140\" | ... 27 more ... | { ...; }",
       "defaultValue": null,
       "required": false,
-      "externalProp": false
+      "externalProp": false,
+      "description": "Responsive prop of Space tokens for the CSS `column-gap` property"
```
[the remark this answers is not in the document]
Comment (SiTaggart) [reply packages/paste-core/components/anchor/type-docs.json:381]: Correct, it's inheriting the descriptions from the style props on box and text

File: packages/paste-core/primitives/box/src/types.ts [hunk 3]
```diff
@@ -32,25 +32,103 @@ export interface BoxStyleProps extends BoxBaseStyleProps, BoxPseudoStyleProps {}
 
 // Omits potential clashes from our style props with HTMLAttributes (i.e.: color)
 export interface BoxElementProps extends Omit<React.HTMLAttributes<HTMLElement>, keyof BoxBaseStyleProps> {
```
[the remark this answers is not in the document]
Comment (SiTaggart) [reply packages/paste-core/primitives/box/src/types.ts:34]: No, not here. Global HTML attributes come with things like `className`, `color`, and `style`. Things we don't allow you to do at the component level.

But here at Box and Text, it's open, free to do what ever you want. It is literally a `div`, whereas when we create a Callout as an example, that is a Paste flavoured `div`

File: packages/paste-core/components/slider/src/Slider.tsx [hunk 4]
```diff
@@ -10,26 +10,121 @@ import { SliderTrack } from "./SliderTrack";
 const DefaultNumberFormatter = new Intl.NumberFormat("en-US");
 
 export interface SliderProps {
```
[the remark this answers is not in the document]
Comment (SiTaggart) [reply packages/paste-core/components/slider/src/Slider.tsx:12]: Good spot.

[the remark this answers is not in the document]
Comment (SiTaggart) [reply packages/paste-core/components/slider/src/Slider.tsx:12]: Hmmm I'm not sure i can here actually. There is a clash between the underlying third party we use and the html input types and I'm not sure I want to touch it in this PR right now. Something we can address in another round as I'm a bit scared to touch it.

**Review by nkrantz — APPROVED** [review]

**Pull request merged**